app/src/main/java/com/android/sample/ui/dialogs/AddImageDialog.kt

```diff
@@ -1,29 +1,81 @@
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
 import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
 
 @Composable
 fun AddImageDialog(onDismiss: () -> Unit, onGalleryClick: () -> Unit, onCameraClick: () -> Unit) {
-  Dialog(onDismissRequest = onDismiss) {
-    Column(modifier = Modifier.fillMaxWidth().padding(16.dp)) {
-      Text(text = "Add an image")
-      Spacer(modifier = Modifier.height(8.dp))
+  Dialog(
+      onDismissRequest = onDismiss,
+      properties =
+          DialogProperties(
+              dismissOnBackPress = true,
+              dismissOnClickOutside = true,
+          )) {
+        Column(
+            modifier =
+                Modifier.fillMaxWidth()
+                    .height(200.dp)
+                    .background(color = Color.White, shape = RoundedCornerShape(size = 16.dp))
+                    .testTag("addUserDialog"),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally,
+        ) {
+          Text(
+              text = "Add an image",
+              style =
+                  TextStyle(color = Color.Black, fontSize = 20.sp, fontWeight = FontWeight.Bold),
+          )
 
-      Button(onClick = onGalleryClick) { Text(text = "Choose from gallery") }
+          Button(onClick = onGalleryClick, modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier.background(Color.Transparent)) {
+                  Text(
+                      text = "Choose from gallery",
+                      style = TextStyle(color = Color.Black, fontSize = 16.sp))
+                  Spacer(modifier = Modifier.width(8.dp))
+                  Icon(Icons.Filled.PhotoLibrary, contentDescription = "Choose from gallery")
+                }
+          }
+          Spacer(modifier = Modifier.height(8.dp))
 
-      Spacer(modifier = Modifier.height(8.dp))
-
-      Button(onClick = onCameraClick) { Text(text = "Take a photo") }
-    }
-  }
+          Button(onClick = onCameraClick, modifier = Modifier.fillMaxWidth().padding(16.dp)) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier.background(Color.Transparent)) {
+                  Text(
+                      text = "Take pictures with camera",
+                      style = TextStyle(color = Color.Black, fontSize = 16.sp))
+                  Spacer(modifier = Modifier.width(8.dp))
+                  Icon(Icons.Filled.PhotoCamera, contentDescription = "Choose from gallery")
+                }
+          }
+        }
+      }
```

Comment:
So for the moment it is still not called in any screens, so this will be implemented on your next pull request.

Reply:
It's called for the moment only on create activity screen but will be implemented on all the screens where we will need the camera